pkg/apis/trust/v1alpha1/types_bundle.go

```diff
@@ -76,6 +76,21 @@ type BundleSource struct {
 	// InLine is a simple string to append as the source data.
 	// +optional
 	InLine *string `json:"inLine,omitempty"`
+
+	// HTTPS downloads a PEM-Formatted trust bundle from an HTTPS server
```

Comment:
We are not _necessarily_ downloading PEM encoded trust bundles. trust is not _currently_ opinionated about the form of the trust bundles- whether they be X.509 CAs, public keys, or just some text.


 ```suggestion
	// HTTPS downloads trust bundles from an HTTPS server
```

Reply:
This comment is ringing major alarm bells for me! I think we _desperately_ need to be aligned on this going forwards. You're of course right about the current state of things.

There are two things here; the formats we accept as inputs and the formats we support as outputs. In either case I don't think it's sustainable  for `cert-manager/trust` to be a controller which essentially implements the unix `cat` command for text files.

I think in an ideal world we should be explicit about what each source accepts as an input, and _always_ validate that those inputs are in a format we expect.

We might want to allow different output formats too. Ideally we should also have exactly one "intermediate" format, to which all inputs can be converted, and from which all outputs can be generated.

If any of that is controversial I think we need to get on a call and make a plan for this or else we need to sit down and agree on a design document on how this would be handled. There's a huge risk of making a mess here I think.